### PR TITLE
[Doc] Add a Go Module

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR adds the `Add a Go Module` previously internal doc to the public one.

### Motivation

This is relevant for public doc.

### CODE REVIEW

File: docs/public/how-to/add/go-module.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+4.  Add `mymodule` to the `DEFAULT_MODULES` in [tasks/modules.py](https://github.com/DataDog/datadog-agent/blob/main/tasks/modules.py)
+    ```
+    DEFAULT_MODULES = (
+    ...,
+    "mymodule": GoModule("mymodule", independent=True, should_tag=False, targets=["."]),
+    )
+    ```
+    - `independent` should it be importable as an independent module ?
+    - `should_tag` should the agent pipeline tag it ?
+    - `targets` should `go test` target specific sub-folders ?
```
Comment (FlorentClarret) [inline docs/public/how-to/add/go-module.md:39]:
```suggestion
    - `independent` should it be importable as an independent module?
    - `should_tag` should the agent pipeline tag it?
    - `targets` should `go test` target specific sub-folders?
```

File: docs/public/how-to/add/go-module.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    package mymodule
+    EOL
+    ```
+
+
+4.  Add `mymodule` to the `DEFAULT_MODULES` in [tasks/modules.py](https://github.com/DataDog/datadog-agent/blob/main/tasks/modules.py)
+    ```
+    DEFAULT_MODULES = (
+    ...,
+    "mymodule": GoModule("mymodule", independent=True, should_tag=False, targets=["."]),
```
Comment (FlorentClarret) [inline docs/public/how-to/add/go-module.md:34]:
```suggestion
    "path/to/mymodule": GoModule("path/to/mymodule", independent=True, should_tag=False, targets=["."]),
```

File: docs/public/how-to/add/go-module.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    )
+    ```
+    - `independent` should it be importable as an independent module ?
+    - `should_tag` should the agent pipeline tag it ?
+    - `targets` should `go test` target specific sub-folders ?
+
+
+5.  If you use your module in another module within `datadog-agent` add `require` and `replace` directives in `go.mod`.
```
Comment (FlorentClarret) [inline docs/public/how-to/add/go-module.md:42]:
```suggestion
5.  If you use your module in another module within `datadog-agent`, add the `require` and `replace` directives in `go.mod`.
```

File: docs/public/how-to/add/go-module.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    From the other module root, Install the dependency with `go get`
```
Comment (FlorentClarret) [inline docs/public/how-to/add/go-module.md:44]:
```suggestion
    From the other module root, install the dependency with `go get`:
```

**Review by FlorentClarret — APPROVED** [review]
Nice! Left some suggestions

File: docs/public/how-to/add/go-module.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
+# Add a Go Module
+
+The repository contains a few submodules. To add a new one and ensure it is tested, follow the following steps:
+
+
+1. Create a directory for the module
```
Comment (jhgilbert) [inline docs/public/how-to/add/go-module.md:6]:
```suggestion
1. Create a directory for the module:
```

File: docs/public/how-to/add/go-module.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    ```
+    cd ~/my_path_to/datadog-agent && mkdir mymodule
+    ```
+
+
+2. Initialise a new go module
```
Comment (jhgilbert) [inline docs/public/how-to/add/go-module.md:12]:
```suggestion
2. Initialize a new Go module:
```

File: docs/public/how-to/add/go-module.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    ```
+    cd mymodule && go mod init
+    ```
+
+
+3.  Create a dummy root root package file `doc.go`:
```
Comment (jhgilbert) [inline docs/public/how-to/add/go-module.md:18]:
```suggestion
3.  Create a dummy root package file `doc.go`:
```

File: docs/public/how-to/add/go-module.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    // Unless explicitly stated otherwise all files in this repository are licensed
+    // under the Apache License Version 2.0.
+    // This product includes software developed at Datadog (https://www.datadoghq.com/).
+    // Copyright 2016-present Datadog, Inc.
+    package mymodule
+    EOL
+    ```
+
+
+4.  Add `mymodule` to the `DEFAULT_MODULES` in [tasks/modules.py](https://github.com/DataDog/datadog-agent/blob/main/tasks/modules.py)
```
Comment (jhgilbert) [inline docs/public/how-to/add/go-module.md:30]:
```suggestion
4.  Add `mymodule` to the `DEFAULT_MODULES` in [tasks/modules.py](https://github.com/DataDog/datadog-agent/blob/main/tasks/modules.py):
```

File: docs/public/how-to/add/go-module.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    ```
+    DEFAULT_MODULES = (
+    ...,
+    "path/to/mymodule": GoModule("path/to/mymodule", independent=True, should_tag=False, targets=["."]),
+    )
+    ```
+    - `independent` should it be importable as an independent module?
```
Comment (jhgilbert) [inline docs/public/how-to/add/go-module.md:37]:
```suggestion
    - `independent`: Should it be importable as an independent module?
```

File: docs/public/how-to/add/go-module.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+    - `should_tag` should the agent pipeline tag it?
```
Comment (jhgilbert) [inline docs/public/how-to/add/go-module.md:38]:
```suggestion
    - `should_tag`: Should the Agent pipeline tag it?
```

File: docs/public/how-to/add/go-module.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+    - `targets` should `go test` target specific sub-folders?
```
Comment (jhgilbert) [inline docs/public/how-to/add/go-module.md:39]:
```suggestion
    - `targets`: Should `go test` target specific subfolders?
```

File: docs/public/how-to/add/go-module.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+
+5.  If you use your module in another module within `datadog-agent`, add the `require` and `replace` directives in `go.mod`.
+
+    From the other module root, install the dependency with `go get`:
+    ```
+    go get github.com/DataDog/datadog-agent/mymodule
+    ```
+    then add the [replace directive](https://go.dev/ref/mod#go-mod-file-replace) in the `go.mod` file:
```
Comment (jhgilbert) [inline docs/public/how-to/add/go-module.md:48]:
```suggestion
    Then add the [replace directive](https://go.dev/ref/mod#go-mod-file-replace) in the `go.mod` file:
```

**Review by jhgilbert — APPROVED** [review]
Approved with minor suggestions for style and a suggestion to remove an extra word. Thank you!

File: docs/public/how-to/add/go-module.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+
+2. Initialise a new go module
+    ```
+    cd mymodule && go mod init
+    ```
+
+
+3.  Create a dummy root root package file `doc.go`:
+    ```
```
Comment (ofek) [inline docs/public/how-to/add/go-module.md:19]:
```suggestion
    ```bash
```

File: docs/public/how-to/add/go-module.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+    // under the Apache License Version 2.0.
+    // This product includes software developed at Datadog (https://www.datadoghq.com/).
+    // Copyright 2016-present Datadog, Inc.
+    package mymodule
+    EOL
+    ```
+
+
+4.  Add `mymodule` to the `DEFAULT_MODULES` in [tasks/modules.py](https://github.com/DataDog/datadog-agent/blob/main/tasks/modules.py)
+    ```
```
Comment (ofek) [inline docs/public/how-to/add/go-module.md:31]:
```suggestion
    ```python
```

File: docs/public/how-to/add/go-module.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+
+5.  If you use your module in another module within `datadog-agent`, add the `require` and `replace` directives in `go.mod`.
+
+    From the other module root, install the dependency with `go get`:
+    ```
+    go get github.com/DataDog/datadog-agent/mymodule
+    ```
+    then add the [replace directive](https://go.dev/ref/mod#go-mod-file-replace) in the `go.mod` file:
+    ```
```
Comment (ofek) [inline docs/public/how-to/add/go-module.md:49]: not yet supported but I opened a feature request https://github.com/pygments/pygments/issues/2748

File: mkdocs.yml [hunk 16]
```diff
@@ -49,6 +49,9 @@ nav:
 - Guidelines:
   - Contributing: guidelines/contributing.md
   - Documentation: guidelines/docs.md
+- How-to:
+  - Add:
+    - A Go module: how-to/add/go-module.md
```
Comment (ofek) [inline mkdocs.yml:54]: what do you think about this?

```suggestion
  - Go:
    - Add module: how-to/go/add-module.md
```

**Review by ofek — DISMISSED** [review]
Awesome, thanks

File: docs/public/how-to/add/go-module.md (hunk 15 above)
Comment (amenasria) [reply docs/public/how-to/add/go-module.md:49]: Nice !

File: mkdocs.yml (hunk 16 above)
Comment (amenasria) [reply mkdocs.yml]: I like it! Fixed in [0fad114](https://github.com/DataDog/datadog-agent/pull/27196/commits/0fad114fc520cb3a9a2a583c1b1c7e3d6e3dd7e8)!

**Pull request merged**